crates/simulacrum/src/store/mod.rs

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::BTreeMap;
+use sui_config::genesis;
+use sui_types::base_types::ObjectRef;
+use sui_types::error::UserInputError;
+use sui_types::transaction::InputObjects;
+use sui_types::transaction::ObjectReadResult;
+use sui_types::transaction::ReceivingObjectReadResult;
+use sui_types::transaction::ReceivingObjects;
+use sui_types::{
+    base_types::{ObjectID, SequenceNumber, SuiAddress},
+    committee::{Committee, EpochId},
+    digests::{ObjectDigest, TransactionDigest, TransactionEventsDigest},
+    effects::{TransactionEffects, TransactionEffectsAPI, TransactionEvents},
+    error::SuiResult,
+    messages_checkpoint::{
+        CheckpointContents, CheckpointContentsDigest, CheckpointDigest, CheckpointSequenceNumber,
+        VerifiedCheckpoint,
+    },
+    object::Object,
+    storage::{BackingStore, ChildObjectResolver, ParentSync},
+    transaction::{InputObjectKind, VerifiedTransaction},
+};
+pub mod in_mem_store;
+
+pub trait SimulatorStore:
+    sui_types::storage::BackingPackageStore
+    + sui_types::storage::ObjectStore
+    + ParentSync
+    + ChildObjectResolver
+{
+    fn init_with_genesis(&mut self, genesis: &genesis::Genesis) {
+        self.insert_checkpoint(genesis.checkpoint());
+        self.insert_checkpoint_contents(genesis.checkpoint_contents().clone());
+        self.insert_committee(genesis.committee().unwrap());
+        self.insert_transaction(VerifiedTransaction::new_unchecked(
+            genesis.transaction().clone(),
+        ));
+        self.insert_transaction_effects(genesis.effects().clone());
+        self.insert_events(
+            genesis.effects().transaction_digest(),
+            genesis.events().clone(),
+        );
+
+        self.insert_to_live_objects(genesis.objects());
+    }
+
+    fn insert_to_live_objects(&mut self, objects: &[Object]);
```

Comment:
Thinking more on this...do we even need this function when we have the `update_objects` function?

Reply:
This is a very good point. On deeper inspection, we can get rid of this function.